src/main/java/hudson/plugins/spotinst/cloud/BaseSpotinstCloud.java

```diff
@@ -643,6 +737,53 @@ protected Integer getPendingThreshold() {
     protected Integer getSlaveOfflineThreshold() {
         return Constants.SLAVE_OFFLINE_THRESHOLD_IN_MINUTES;
     }
+
+    public void handleGroupManagedByOtherController() {
+        handleInitializingExpired();
+
+        if (groupAcquiringDetails.getState().equals(SPOTINST_CLOUD_COMMUNICATION_READY)) {
+            groupAcquiringDetails = new GroupAcquiringDetails(groupId, accountId);
+        }
+    }
+
+    public void syncGroupOwner() {
+        ILockRepo           lockRepo                    = RepoManager.getInstance().getLockRepo();
+        ApiResponse<String> lockGroupControllerResponse = lockRepo.getGroupControllerLockValue(accountId, groupId);
+
+        if (lockGroupControllerResponse.isRequestSucceed()) {
+            String  lockGroupControllerValue       = lockGroupControllerResponse.getValue();
+            String  currentControllerIdentifier    = SpotinstContext.getInstance().getControllerIdentifier();
+            boolean isGroupAlreadyHasAnyController = lockGroupControllerValue != null;
+
+            if (isGroupAlreadyHasAnyController) {
+                boolean isGroupBelongToCurrentController = currentControllerIdentifier.equals(lockGroupControllerValue);
+
+                if (isGroupBelongToCurrentController) {
+                    ExpandGroupLock(currentControllerIdentifier);
+                }
+                else {
+                    LOGGER.warn(
```

Comment:
move it into handleGroupManagedByOtherController , then you can decide whether it error or warning

Reply:
done